### PR TITLE
JAMES-2855 Upgrade build maven plugins

### PR DESCRIPTION
**Pull request by chibenwa**
```
Before (mvn clean install)
[INFO] Total time:  06:28 min (Wall Clock)

After
[INFO] Total time:  05:36 min (Wall Clock)
```

Nice enhancement ;-)

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[2d1b8d5c5dc8589f234e4fa71efa3ef4cb247aea] [ERROR] Failed to execute goal on project apache-james-mailbox-spring: Could not resolve dependencies for project org.apache.james:apache-james-mailbox-spring:jar:3.4.0-SNAPSHOT: Failed to collect dependencies at commons-dbcp:commons-dbcp:jar:1.4: Failed to read artifact descriptor for commons-dbcp:commons-dbcp:jar:1.4: Could not transfer artifact commons-dbcp:commons-dbcp:pom:1.4 from/to central (https://repo.maven.apache.org/maven2): Connection reset
```

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[2d1b8d5c5dc8589f234e4fa71efa3ef4cb247aea] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
[2d1b8d5c5dc8589f234e4fa71efa3ef4cb247aea] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
```

**Comment by chibenwa** [comment]
test this please

File: pom.xml [hunk 1]
```diff
@@ -650,6 +650,8 @@
 
         <scala.base>2.12</scala.base>
         <scala.version>${scala.base}.8</scala.version>
+
+        <doclint>none</doclint>
```
Comment (rouazana) [inline pom.xml]: usually with have this:
```
        <profile>
            <id>doclint-java8-disable</id>
            <activation>
                <jdk>[1.8,)</jdk>
            </activation>
            <build>
                <plugins>
                    <plugin>
                        <groupId>org.apache.maven.plugins</groupId>
                        <artifactId>maven-javadoc-plugin</artifactId>
                        <configuration>
                            <additionalparam>-Xdoclint:none</additionalparam>
                        </configuration>
                    </plugin>
                </plugins>
            </build>
        </profile>
```
any reason to change? Or then not to remove old doclint way?

File: mpt/core/src/main/java/org/apache/james/mpt/protocol/FileProtocolSessionBuilder.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
 /**
  * A builder which generates a ProtocolSession from a test file.
  * 
- * @author Darrell DeBoer <darrell@apache.org>
+ * @author Darrell DeBoer &lt;darrell@apache.org&gt;
```
Comment (mbaechler) [inline mpt/core/src/main/java/org/apache/james/mpt/protocol/FileProtocolSessionBuilder.java]: author tags are discouraged by Apache rules AFAIK

File: pom.xml (hunk 1 above)
Comment (chibenwa) [reply pom.xml]: Well we enforce the build to be java 11, thus why using some 'if(java 8) then xxx' ?

It adds complexity, and duplicated javadoc configuration (one in the profile for java 8 and more, one for non java 8 in the plugin config).

Comment (rouazana) [reply pom.xml]: Fine, then you can resolve the last part of my comment: remove the old doclint way.
```
mpt/app/pom.xml
mpt/core/pom.xml
mpt/mavenplugin/pom.xml
```

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**